src/Polysemy/Internal.hs

```diff
@@ -351,8 +356,8 @@ raiseUnder :: ∀ e2 e1 r a. Sem (e1 ': r) a -> Sem (e1 ': e2 ': r) a
 raiseUnder = hoistSem $ hoist raiseUnder . weakenUnder
   where
     weakenUnder :: ∀ m x. Union (e1 ': r) m x -> Union (e1 ': e2 ': r) m x
-    weakenUnder (Union SZ a) = Union SZ a
-    weakenUnder (Union (SS n) a) = Union (SS (SS n)) a
+    weakenUnder (Union Here a) = Union Here a
```

Comment:
Do we have a way to generalize this with the new machinery? I've always hated having to write a few blessed `raise` functions.

Reply:
No, there's nothing that the new membership proof machinery can help on this point.

This should be implementable, though:
```haskell
type family Underneath n r e where
  Underneath 'Z r e = e ': r
  Underneath ('S n) (x ': r) e = x ': Underneath n r e

raiseUnderX :: SNat n -> Sem r a -> Sem (Underneath n r e) a
```